pkg/ccpatch/substitution/ipv4address/substitution_ipv4_address.go

```diff
@@ -0,0 +1,79 @@
+package ccpatch
+
+import (
+	"fmt"
+	"net"
+	"slices"
+
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/pkg/ccpatch/ipnet"
+	"github.com/ionos-cloud/crossplane-provider-ionoscloud/pkg/ccpatch/substitution"
+)
+
+func init() {
+	substitution.RegisterSubstitution(&ipv4Address{})
+}
+
+type ipv4Address struct{}
+
+var _ substitution.Handler = &ipv4Address{}
+
+func (i *ipv4Address) Type() string {
+	return "ipv4Address"
+}
+
+func (i *ipv4Address) WriteState(identifier substitution.Identifier, gs *substitution.GlobalState, sub substitution.Substitution) error {
+	value, ok := sub.AdditionalProperties["cidr"]
+	if !ok {
+		return substitution.ErrMissingCIDR
+	}
+
+	if gs.Exists(identifier, sub.Key) {
+		return nil
+	}
+
+	used := []string{}
+
+	if sub.Unique {
+		gs.Each(func(key substitution.Identifier, state []substitution.State) {
+			for _, s := range state {
+				if s.Key == sub.Key {
+					used = append(used, s.Value)
```

Comment:
here the capacity of the slice will be recalculated a few times in that for loop

Reply:
you can prealloc the slice on line 35(?)

Reply:
Yes with the size of the `en(state)`, because in the fortunate scenario all the state keys might match the sub.key

Reply:
Good idea, I don't think we will have a large number of substitutions and the append function will double the size of the underlying array for lower value and then 1.25 for higher where allocation gets expensive. Will leave this as is, until we see that it works slowly and is too expensive.

Reply:
Ok, let's do it as such